salt/utils/path.py

```diff
@@ -344,11 +239,10 @@ def join(*parts, **kwargs):
     The "use_posixpath" kwarg can be be used to force joining using poxixpath,
     which is useful for Salt fileserver paths on Windows masters.
     """
-    if six.PY3:
-        new_parts = []
-        for part in parts:
-            new_parts.append(salt.utils.stringutils.to_str(part))
-        parts = new_parts
+    new_parts = []
+    for part in parts:
+        new_parts.append(salt.utils.stringutils.to_str(part))
+    parts = new_parts
```

Comment:
**suggestion** not necessarily part of this change, but just `parts = [salt.utils.stringutils.to_str(part) for part in parts]` would do it.